src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -444,6 +452,61 @@ private void SetTransferMode(TransferMode transferMode)
             _httpClient.Dispose();
         }
 
+        public IDisposable OnReceived(Func<byte[], object, Task> callback, object state)
+        {
+            var callBack = new ReceiveCallBack(callback, state);
+            lock (_callBacks)
+            {
+                _callBacks.Add(callBack);
+            }
+            return new Subscription(callBack, _callBacks);
+        }
+
+        public IDisposable OnReceived(Func<byte[], Task> callback)
```

Comment:
I don't think this should be on HttpConnection - this is just syntactic sugar.

Reply:
Should I make a extensions file for just this method?

Reply:
Yes.